### DISCUSSION TITLE
Make network speed widget a fixed size

### DISCUSSION THREAD

**Discussion by RedBearAK** [JaKooLit/Hyprland-Dots#99] [Ideas]
On the Waybar style I've been using, there is a network info widget that keeps changing its width as the number of digits in the network speeds changes. This makes it very difficult to interact with the system tray icons. 

I'm sure there's a simple way to disable it entirely, but if there is a way to force that widget to adopt a fixed size, that would be nice. 

Very minor issue.

**Comment by RedBearAK** [comment]
Actually it looks like some of the other widgets like the CPU percentage on the left do the same kind of thing. But it's more of a problem with the network widget, since the tray icons are on its left side and keep getting moved around.

**Comment by JaKooLit** [comment]
its not an issue but thats because the the network keep changing... you can remove that module or choose another waybar layout 

super ctrl b layouts and super alt b for styles

that network widget is only on the default layout and not present on any waybar layout

you can costumize the modules as much as you wanted

**Comment by RedBearAK** [comment]
@JaKooLit 

> you can costumize the modules as much as you wanted

True. There are an overwhelming number of choices. Would be nice to have a gallery of images that showcases all the waybar layouts and styles. I scanned through the README and didn't see anything like that.

**Comment by JaKooLit** [comment]
hmmmmmm I can but that would mean like more than more than hundreds of screenshots....

cause any styles, you can combine or apply to any layout........

**Comment by JaKooLit** [comment]
https://github.com/JaKooLit/Hyprland-Dots/wiki

maybe you want to take a look here as well...

**Comment by RedBearAK** [comment]
> more than more than hundreds of screenshots

No, just pick a default style and show all the different layouts, then show the styles separately with a single layout. You wouldn't need to show all the possible combinations. People will be able to figure out how to pair the layout they want with the style they like. But just going by name and having to try them all to see them and then remember which ones you liked, it's a bit hard. Especially if you have ADHD and a poor working memory.

**Comment by JaKooLit** [comment]
![Screenshot_2023-12-22-00-35-21_32375](https://github.com/JaKooLit/Fedora-Hyprland/assets/85185940/df6ba38d-da24-46f9-a64c-d10ee5b55c8d)

**Comment by RedBearAK** [comment]
I can't find the original thread about Hyprland using too much CPU in the virtual machine, but I wanted to update you. Turns out it was user error. The 3D acceleration and OpenGL switches that I enabled somehow got disabled again when running the VM with GNOME Boxes, but when I went back into Virt-Manager and made sure they were enabled, the CPU usage goes down to 1-2% instead of 85%. So Hyprland just isn't very usable without the 3D acceleration enabled. My mistake with the VM settings. 

Looking forward to installing a new iteration of the dot files. 👍🏽

**Reply by JaKooLit** [reply]
I moved it here

thought github will take care of the the notification part

https://github.com/JaKooLit/Hyprland-Dots/discussions/86

**Comment by JaKooLit** [comment]
> Looking forward to installing a new iteration of the dot files. 👍🏽

You can try now here. Please run the copy.sh
https://github.com/JaKooLit/Hyprland-Dots

Please remember to double check the backups created...

Especially the ~/Pictures/wallpapers

**Comment by RedBearAK** [comment]
Tried it, and the moving of the network widget is a very welcome change. Thanks! I still think all the widgets with changing numbers of digits may have configuration options to cause them to have leading zeros or spaces so they can be more of a static size, but I don't have time to look into how to do that. 

When I was looking over the README for the dot files before, I didn't realize there was a "copy" option. I was in a hurry and just scanned the README and didn't find an "install" section, so to get the dot files updated previously I just re-ran the whole Fedora-Hyprland script. Maybe to catch the eye you could add a header in the dot files README that says something about "how to install/update", where you are currently talking about the copy/release scripts. 

Great work.

**Comment by JaKooLit** [comment]
@RedBearAK  discussions moved here

> Tried it, and the moving of the network widget is a very welcome change. Thanks! I still think all the widgets with changing numbers of digits may have configuration options to cause them to have leading zeros or spaces so they can be more of a static size, but I don't have time to look into how to do that.

Reviewed the waybar wiki and only can set a maximum length

> 
> When I was looking over the README for the dot files before, I didn't realize there was a "copy" option. I was in a hurry and just scanned the README and didn't find an "install" section, so to get the dot files updated previously I just re-ran the whole Fedora-Hyprland script. Maybe to catch the eye you could add a header in the dot files README that says something about "how to install/update", where you are currently talking about the copy/release scripts.
> 
> Great work.

- Readme updated

**Comment by RedBearAK** [comment]
> Reviewed the waybar wiki and only can set a maximum length

Huh. Not how I would have designed the modules, but that's just me. It is what it is. 

> Readme updated

Looks great. 👍🏽 Thanks.

**Comment by nr23730** [comment]
I replaced the network module with this one: https://github.com/Alexays/Waybar/issues/2162
This one has a fixed width.
